fog/view/server/tests/router_smoke_tests.rs

```diff
@@ -93,28 +218,210 @@ async fn test_router_integration(test_environment: &mut RouterTestEnvironment, l
+        assert_eq!(sort_txs[0].search_key, vec![1u8; 16]);
+        assert_eq!(sort_txs[0].result_code, 1);
+        assert_eq!(sort_txs[0].ciphertext, vec![1u8; 232]);
+
+        assert_eq!(sort_txs[1].search_key, vec![2u8; 16]);
+        assert_eq!(sort_txs[1].result_code, 1);
+        assert_eq!(sort_txs[1].ciphertext, vec![2u8; 232]);
+
+        assert_eq!(sort_txs[2].search_key, vec![3u8; 16]);
+        assert_eq!(sort_txs[2].result_code, 1);
+        assert_eq!(sort_txs[2].ciphertext, vec![3u8; 232]);
+    }
+    assert_eq!(result.missed_block_ranges.len(), 0); // no range reported since we started at event id 4
+    assert_eq!(result.last_known_block_count, 6);
+
+    let result = router_client
+        .query(4, 0, vec![vec![5u8; 16], vec![8u8; 16], vec![200u8; 16]])
+        .await;
+    assert!(result.is_ok());
+    let result = result.unwrap();
+    assert_eq!(result.highest_processed_block_count, 6);
+    assert_eq!(result.next_start_from_user_event_id, 4);
+    assert_eq!(result.rng_records.len(), 0);
+    assert_eq!(result.tx_out_search_results.len(), 3);
+    {
+        let sort_txs = interpret_tx_out_results(result.tx_out_search_results.clone());
+        assert_eq!(sort_txs[0].search_key, vec![5u8; 16]);
+        assert_eq!(sort_txs[0].result_code, 1);
+        assert_eq!(sort_txs[0].ciphertext, vec![5u8; 232]);
+
+        assert_eq!(sort_txs[1].search_key, vec![8u8; 16]);
+        assert_eq!(sort_txs[1].result_code, 1);
+        assert_eq!(sort_txs[1].ciphertext, vec![8u8; 232]);
+
+        assert_eq!(sort_txs[2].search_key, vec![200u8; 16]);
+        assert_eq!(sort_txs[2].result_code, 2);
+        // FIGURE OUT WHAT TO DO HERE
+        assert_eq!(sort_txs[2].ciphertext, vec![0u8; 254]);
+    }
+
+    assert_eq!(result.missed_block_ranges.len(), 0); // no range reported since we started at event id 4
+    assert_eq!(result.last_known_block_count, 6);
+
+    let result = router_client.query(0, 0, vec![vec![200u8; 17]]).await;
+    assert!(result.is_ok());
+    let mut result = result.unwrap();
+    result
+        .rng_records
+        .sort_by_key(|rng_record| rng_record.ingest_invocation_id);
+    assert_eq!(result.highest_processed_block_count, 6);
+    assert_eq!(result.next_start_from_user_event_id, 4);
+    assert_eq!(result.rng_records.len(), 2);
+    assert_eq!(result.rng_records[0].pubkey, egress_public_key_1);
+    assert_eq!(result.rng_records[1].pubkey, egress_public_key_2);
+    assert_eq!(result.tx_out_search_results.len(), 1);
+    {
+        let sort_txs = interpret_tx_out_results(result.tx_out_search_results.clone());
+        assert_eq!(sort_txs[0].search_key, vec![200u8; 17]);
+        assert_eq!(sort_txs[0].result_code, 3);
+        assert_eq!(sort_txs[0].ciphertext, vec![0u8; 232]);
+    }
+    assert_eq!(result.missed_block_ranges.len(), 1);
+    assert_eq!(result.missed_block_ranges[0], BlockRange::new(3, 4));
+    assert_eq!(result.last_known_block_count, 6);
+}
```

Comment:
While this does simplify the logic, I take issue with what happens when the test fails. I just ran the tests and intentionally changed some parameters to make the test fail, and the error message is something like: `left didn't equal right at fog/view/server/tests/router_smoke_tests.rs:395:5`. This line number corresponds to the new assert method, and doesn't give any information about which part of the test failed. I would now have to go though the test and add log statements or start up CLion and run a debugger to see exactly where the test failed. 

IMO, the difficulty in debugging a test written in this way outweighs the cleanliness and I'd rather keep it the way it is. However, maybe I'm overstating the debugging difficulty and would change it if you feel strongly.

Reply:
> This line number corresponds to the new assert method
this is one of the downsides of moving the logic into a support function.  I'm fine leaving this as is, only time will tell if spending cycles trying to make this test more concise will be beneficial.

Not saying to do it here, but I wonder if this would be a good use case for a declarative macro.  It could be that, https://www.youtube.com/watch?v=q6paRBbLgNw is my current lunch time bite size video so it happens to be on my mind

Reply:
I think this is a good idea, and I'll definitely implement it in the future once we get the release out.